.github/workflows/anaconda_pkg_build_linux_cuda_rocky8.yml

```
@@ -0,0 +1,76 @@
+name: Build and publish Linux CUDA package builder images
+on:
+  #push:
+  #  branches:
+  #    - main
+  #  tags:
+  #    - 'pkg-build-*'
+  #  paths:
+  #    - 'anaconda-pkg-build/linux/cuda-rocky8/Dockerfile'
+  #    - '.github/workflows/anaconda_pkg_build_linux_cuda_rocky8.yml'
+  #pull_request:
+  #  paths:
+  #    - 'anaconda-pkg-build/linux/cuda-rocky8/Dockerfile'
+  #    - '.github/workflows/anaconda_pkg_build_linux_cuda_rocky8.yml'
```

There is no prove that this PR works... please enable so that it at least runs when the workflow changes in PRs:
```suggestion
  pull_request:
    paths:
      - 'anaconda-pkg-build/linux/cuda-rocky8/Dockerfile'
      - '.github/workflows/anaconda_pkg_build_linux_cuda_rocky8.yml'
```

---

leaving this commented out was intentional: I didn't want it to be rebuilt and uploaded upon each PR push, or even merge to main, only when manually triggered. I had to [re-comment it for a different dockerfile](https://github.com/anaconda/docker-images/pull/629) for that reason.
Also, I'm looking for approval to actually upload the docker image publicly, before I can run this action. 
Would you still suggest running this action and uploading it as part of this PR's development? Or would you suggest something else?

---

* CI/CD ftw... without enabling it there is no prove that this workflow actually works 
* the workflow only really pushes under those conditions `push: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags') }}` ... so enabling PR builds is now harm and makes reviewing changes to the workflow or Dockerfile easier
* while at it: please also enable the tag trigger and tag a release after PR merge... and only use tagged images in production.

---

> please also enable the tag trigger and tag a release after PR merge... and only use tagged images in production.

Thanks, so concretely would that mean removing the "push on merge to main" behaviour? i.e.

`push: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags') }}`

changes to

`push: ${{ startsWith(github.ref, 'refs/tags') }}`

---

ok thanks. so:
- during a PR's development - images built, not uploaded
- upon merge - images built and uploaded but not tagged (shouldn't use in production)
- upon git tagging of a release commit on main, and upload - images built and docker-tagged and uploaded (should use in production)
Thanks for the explanation, still learning about typical gh action workflows.

Would you be able to have a look at the PR again, is it ok now?